cfme/infrastructure/provider/kubevirt.py

```diff
@@ -0,0 +1,45 @@
+from cfme.utils.providers import get_crud
+from cfme.containers.provider.openshift import VirtualizationEndpoint
+from . import InfraProvider
+
+
+class KubeVirtProvider(InfraProvider):
+
+    def __init__(self, name=None, endpoints=None, key=None, provider_data=None,
+                 parent_provider=None, appliance=None):
+
+        self.parent_provider = parent_provider
+
+        super(KubeVirtProvider, self).__init__(
+            name=name,
+            endpoints=endpoints,
+            key=key,
+            provider_data=provider_data,
+            appliance=appliance)
+
+    def create(self, *args, **kwargs):
+        # Override the standard behaviour to actually create the parent container provider first.
+        created = self.parent_provider.create()
```

Comment:
This blows up with AttributeError when parent_provider is `None`

Reply:
Pass `**kwargs` on to the create() method so that the `BaseProvider.create()` kwargs can be passed through this `cancel=False, validate_credentials=True, check_existing=False, validate_inventory=False`

Reply:
I feel `AttributeError` would be okay here as parent_provider must be there, or can just handle is with parent_provider check and raise an exception

Reply:
If you made the parent_provider a required kwarg, then its a non-issue :)

Reply:
yup made it a required kwarg